packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -15,11 +15,16 @@ import { t } from '../i18n';
  * Sets an Etag which will be used for re-validation of the cached UI5 sources.
  *
  * @param proxyRes - proxy response object
+ * @param req
  * @param etag - ETag for the cached sources, normally the UI5 version
  */
-export const proxyResponseHandler = (proxyRes: IncomingMessage, etag: string): void => {
+export const proxyResponseHandler = (proxyRes: IncomingMessage, req: Request, etag: string): void => {
     proxyRes.headers['Etag'] = etag;
     proxyRes.headers['cache-control'] = 'no-cache';
+    if (proxyRes.statusCode === 404 && req.path.includes('/resources/sap/ushell/bootstrap/sandbox2.js')) {
```

Comment:
Not sure if this is relevant here but please ensure that this also works when preview middleware is used with karma (uses connect API instead of express). See [karma-ui5 documentation](https://github.com/SAP/karma-ui5?tab=readme-ov-file#notes-on-ui5-tooling-custom-server-middleware) (and #1968 and #1826)

Reply:
@heimwege 
Hello Dominik, it does not look like this coding enhancement contradicts to the statements and PR's concerning karma, but if you can test it it would be helpful. Or let me do about the right test environment that I can use.

Reply:
Sure, I'll drop you a mail as the repo is internal